lib/modifiers.js

```diff
@@ -200,35 +195,66 @@ function iterateNthLevel (wcov, level, k, path, afterPath, censor, isCensorFct,
           ? (censorFctTakesPath ? censor(ov, [...path, originalKey, ...afterPath]) : censor(ov))
           : censor)
       if (kIsWc) {
+        const rv = restoreInstr(redactPathCurrent, ov, parent)
+        store.push(rv)
         n[wck] = nv
       } else {
         if (wcov[k] === nv) {
-          exists = false
+          // pass
+        } else if ((nv === undefined && censor !== undefined) || (has(wcov, k) && nv === ov)) {
+          // pass
         } else {
-          wcov[k] = (nv === undefined && censor !== undefined) || (has(wcov, k) && nv === ov) ? wcov[k] : nv
+          const rv = restoreInstr(node(redactPathCurrent, k, depth + 1), ov, parent)
+          store.push(rv)
+          wcov[k] = nv
         }
       }
     }
-    return ov
   }
   for (const key in wcov) {
     if (typeof wcov[key] === 'object') {
-      var temp = iterateNthLevel(wcov[key], level - 1, k, path, afterPath, censor, isCensorFct, censorFctTakesPath, originalKey, n, nv, ov, kIsWc, wck, i, lastPathIndex, exists)
-      return temp
+      redactPathCurrent = node(redactPathCurrent, key, depth)
+      iterateNthLevel(wcov[key], level - 1, k, path, afterPath, censor, isCensorFct, censorFctTakesPath, originalKey, n, nv, ov, kIsWc, wck, i, lastPathIndex, redactPathCurrent, store, parent, depth + 1)
     }
   }
 }
 
-function restoreNthLevel (key, target, value, level) {
-  if (level === 0) {
-    if (has(target, key)) {
-      target[key] = value
-    }
-    return
+function tree () {
+  return { parent: null, key: null, children: [], depth: 0 }
+}
+
+/**
+ * creates a new node in the tree, attaching it as a child of the provided parent node
+ * if the specified depth matches the parent depth, adds the new node as a _sibling_ of the parent instead
+ */
+function node (parent, key, depth) {
```

Comment:
Thank you for adding some jsdoc. Please complete it by adding the `@param` and `@returns` directives.